src/trust/chain.ts

```diff
@@ -107,3 +108,40 @@ export async function verifyTrustChain(
       .map((args) => verify(...args))
   );
 }
+
+/**
+ * Given a trust chain, obtain a new trust chain by fetching each element's fresh version
+ *
+ * @param chain The original chain
+ * @param appFetch (optional) fetch api implementation
+ * @returns A list of signed token that reprensent the trust chain, in the same order of the provided chain
+ * @throws When an element of the chain fails to parse
+ */
+export function renewTrustChain(
+  chain: string[],
+  appFetch: GlobalFetch["fetch"] = fetch
+) {
+  return Promise.all(
+    chain
+      // Decode each item to determine its shape
+      .map(decode)
+      .map(
+        (e) =>
+          [
+            EntityStatement.safeParse(e),
+            EntityConfiguration.safeParse(e),
+          ] as const
+      )
+      // fetch the element accorfing to its shape
+      .map(([es, ec], i) =>
+        ec.success
+          ? getSignedEntityConfiguration(ec.data.payload.iss, { appFetch })
+          : es.success
+          ? getSignedEntityStatement(es.data.payload.iss, es.data.payload.sub, {
+              appFetch,
+            })
+          : // if the element fail to parse in both EntityStatement and EntityConfiguration, raise an error
+            Promise.reject(`Failed to parse element #${i} of the trust chain`)
```

Comment:
Here you could use the `Error` object common to the entire library.

Reply:
Fixed in https://github.com/pagopa/io-react-native-wallet/pull/52/commits/be1ddbe72fcc8c7015a832063d52a608399d312f